enigma-js/src/enigma-utils.js

```diff
@@ -210,7 +210,7 @@ function appendMessages(hexStr, inputsArray, principal=false) {
 function appendArrayMessages(hexStr, inputsArray, principal=false) {
   const principalPrefix = principal ? '01' : '';
   for (let array of inputsArray) {
-    hexStr += principalPrefix + JSBI.BigInt(array[0].length * array[1]).toString(16).padStart(16, '0');
+    hexStr += principalPrefix + JSBI.BigInt(array[0].length * (array[1]+9)).toString(16).padStart(16, '0');
```

Comment:
I am not sure that this will work good if `principal` is false. Is this code relevant if `principal` is false?

Reply:
it's in place for the time being as a placeholder for when our network supports batching of commit receipts

Reply:
To put @apalepu23's answer in context, we did not have clarity whether the new hashing method is going to be used anywhere where an array is involved (this should be the desired choice), or only in the hashing of KM operations (that necessarily involve arrays).

Thus, this `if` around `principal` is to account for that uncertainty; which I would rather remove after confirming that the new hashing method will apply everywhere, not only for the KM.

I am thus seeking confirmation/clarification from @moriaab, @AvishaiW before merging.

Reply:
It should apply everywhere *in the future*, currently we implemented this only for the KM node

Reply:
Correct. And from this it follows, that *currently* there is nowhere else where hashing of arrays take place; so we can remove this `if`. Thanks

Reply:
Please note that the current implementation calls for adding `00` to individual values that will be hashed as part of the KM hashing algorithm where there are also arrays, but core hashes individual values without the `00` prefix.

Reply:
I am completely agree with removing the `if`.

Reply:
@lacabra, regarding adding `00` you mentioned. There is an inconsistency indeed, but we can live with it. The purpose of the hashing algorithm is to do a unique serialization for an input values. When it is known that only individual values can appear in the input, it is meaningless to add `00` to express this.

Reply:
@moriaab thanks for the clarification (Cc: @apalepu23). All clear now.
That `if` was indeed removed.